app/controllers/employees_controller.rb

```diff
@@ -0,0 +1,59 @@
+class EmployeesController < ApplicationController
+  
+  before_action :set_employee, only: [:show, :edit, :update, :destroy]
+  
+  def index
+    @department_id = params[:department_id]
+    if params[:department_id].present?
+      @department = Department.find(params[:department_id])
+      @employees = @department.employees
+    else
+      @employees = Employee.all
+    end
+  end
+
+  def show
+  end
+
+  def new
+    @employee = Employee.new
+  end
+
+  def create
+    @employee = Employee.new(employee_params)
```

Comment:
I don't get it... Can you provide some examples of code how it works?

Reply:
I meant no need for `@employee` since you don't use it anywhere except this method. You can just write `employee = ...`. `employee` takes less memory then `@employee`.